### PR TITLE
boards: lpcxpresso54114: Combine m4/m0 boards into one directory

### PR DESCRIPTION
**Pull request by MaureenHelm**
Combines the lpcxpresso54114_m4 and lpcxpresso54114_m0 boards into one
directory containing both board configurations. This eliminates some
duplicate code and documentation, and better represents the actual
hardware.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

@MeganHansen This will change the url for the board document. We currently have two board documents (one for each core):
1. https://docs.zephyrproject.org/latest/boards/arm/lpcxpresso54114_m4/doc/lpcxpresso54114.html
2. https://docs.zephyrproject.org/latest/boards/arm/lpcxpresso54114_m0/doc/lpcxpresso54114_m0.html

This change will combine them into one:
https://docs.zephyrproject.org/latest/boards/arm/lpcxpresso54114/doc/lpcxpresso54114.html

### CODE REVIEW

File: boards/arm/lpcxpresso54114/lpcxpresso54114.dtsi [hunk 1]
```diff
@@ -20,12 +13,6 @@
 		led2 = &blue_led;
 	};
 
-	chosen {
-		zephyr,sram = &sram0;
-		zephyr,flash = &flash0;
-		zephyr,console = &usart0;
-	};
-
 	leds {
```
Comment (galak) [inline boards/arm/lpcxpresso54114/lpcxpresso54114.dtsi]: Having the LEDs shared between both boards seems problematic.  Not sure if it makes sense to have all the LEDs on one core, or split them.

**Review by galak — CHANGES_REQUESTED** [review]
In general looks good, will let @dbkinder comment on the doc merge.  Just the sharing of LEDs is my only comment.

**Review by dbkinder — APPROVED** [review]
works for me.  the previous separation of the docs based on which core you were working with seemed awkward.

**Review by ioannisg — COMMENTED** [review]
Looks good to me, too.
I wonder if we really want to have a .dtsi extension in DTS files in board definition (we traditionally have .dts files)

File: boards/arm/lpcxpresso54114/lpcxpresso54114.dtsi (hunk 1 above)
Comment (MaureenHelm) [reply boards/arm/lpcxpresso54114/lpcxpresso54114.dtsi]: It's the same problem as having a uart instance available to both cores on the soc. How about we handle the leds the same way, where we define them in the shared dtsi file with `status = "disabled"`, and then the core-specific dts can choose to enable them. This way we can enable them by default for the m4, but allow an override via a dts overlay if an application wants to use them in the m0 instead.

Comment (galak) [reply boards/arm/lpcxpresso54114/lpcxpresso54114.dtsi]: That works, just didn't want the default case to have both M0 & M4 accessing them.

Comment (MaureenHelm) [reply boards/arm/lpcxpresso54114/lpcxpresso54114.dtsi]: Done

**Review by galak — APPROVED** [review]

**Pull request merged**